### DISCUSSION TITLE
Don't have rc hash? Signage point xx not added???

### DISCUSSION THREAD

**Discussion by pluto2021** [Chia-Network/chia-blockchain#4126] [Q&A]
Recently getting a lot of these messages in the log. Wonder if anyone knows what they mean?

2021-05-07T15:24:13.513 full_node chia.full_node.full_node_store: INFO     Don't have rc hash b5cf9xxxxxxxxd3b5. caching signage point 20.
2021-05-07T15:24:13.514 full_node chia.full_node.full_node: INFO     Signage point 20 not added, CC challenge: c226bxxxxxx10c5, RC challenge: b5cf9xxxxxxxxxd3b5

**Comment by andre-bender** [comment]
I am getting the same messages.

**Comment by Nikolaj-K** [comment]
From looking at the code, I can at least tell you that it's from [this function](https://github.com/trentontemple/chia-blockchain/blob/f18eba5aca6dc6f74adda849285b4cc939d2cff5/chia/full_node/full_node_store.py#L183). It's always gonna log if the `signage_point` aren't `None`, so you can probably reverse engineer where it's called and fails in this file.
(Someone should refactor the broken English in the docs and log btw., there seems to be no standard in place. Or whether to write full sentences or not - this one interestingly ends with a `.`, but doesn't start capitalized.)

**Reply by pluto2021** [reply]
Any response from the developer team on this question? Something is not quite right. Need clear understanding of the nature of this log message.

**Comment by mariano54** [comment]
This is fine, it means you received the signage point before you received the block that it depends on.

**Reply by NormanVelez** [reply]
i have been strugling for 2 days and i am missing a lot of block opportunities due to the bugs in version 1.2.10 that chia is claiming is not the issue. when will next version with bug fix be released?

**Comment by BrandtH22** [comment]
If you are still experiencing the issues it is recommended to upgrade to the latest version of chia. Please download it from here: https://www.chia.net/downloads/

Since this issue has been open for a number of years without additional comments, we will be closing this ticket but if we have closed this ticket in error do not hesitate to reach out to us again with any followup questions, comments, or if the issue persists after an update.

```
Please note that upgrading major versions of chia generally requires the config.yaml to be recreated using these steps:
1. Stop chia
2. Rename the config or move it to a different directory
3. In CLI run chia init
4. Copy any custom settings from the old config to the new (plot drive directories, pool info, trusted node info, etc) - be careful here as yaml is very picky about formatting including those leading spaces
5. Start Chia
```

The best place to reach our support team is on Discord (https://discord.gg/chia) or by reopening this ticket.
